src/nvim/ex_docmd.c

```diff
@@ -6382,6 +6382,10 @@ ex_win_close(
   int need_hide;
   buf_T       *buf = win->w_buffer;
 
+  if (!buf) {
```

Comment:
might as well check `buf_valid` instead.

Reply:
Apparently I lied about adding the test, but yes I will actually do it today, and check out buf_valid too.